### PR TITLE
Implement filtering by income

### LINKED ISSUE #64: Support filtering income based on greater/less/equal

### PR DESCRIPTION
**Pull request by itsme-zeix**
Allows users to filter customers by income using operators (`=`, `>` or `<`).

See commits for changes.

Closes #64.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -306,6 +307,19 @@ To perform a search, use the `filter` command followed by one or more flags (ind
...
+
+#### Filtering By Income
+- **Comparison Operators:** Filtering by income allows numeric comparisons using operators `=`, `>`, or `<` to find customers whose income meets certain criteria.
+- **Equal to (`=`):** Use `=` to find customers with a specific income.  
+  - `i/ =5000` will match customers with an income of exactly 5000.
+- **Greater than (`>`):** Use `>` to find customers with an income higher than the specified threshold. 
+  - `i/ >5000` will match customers with incomes greater than 5000.
+- **Less than (`<`):** Use `<` to find customers with an income lower than the specified threshold.
+  - `i/ <5000` will match customers with incomes below 5000.
+
```
Comment (iamdiluxedbutcooler) [inline docs/UserGuide.md:322]: Thank for updating the UG!

File: src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private final IncomeComparisonOperator incomeComparisonOperator;
+
+    /**
+     * Constructs an {@code IncomeComparisonPredicate}.
+     *
+     * @param incomeComparisonOperator The operator used to compare the person's income with the threshold.
+     * @param incomeThreshold          The threshold income to compare against.
+     */
+    public IncomeComparisonPredicate(IncomeComparisonOperator incomeComparisonOperator, int incomeThreshold) {
+        requireNonNull(incomeComparisonOperator);
```
Comment (iamdiluxedbutcooler) [inline src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java:25]: Good job on using `requireNonNull` to validate the `incomeComparisonOperator`. Have you considered adding a similar check for `incomeThreshold`? (to ensure it's not negative)

File: src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        this.incomeThreshold = incomeThreshold;
+        this.incomeComparisonOperator = incomeComparisonOperator;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        int personIncome = person.getIncome().value;
+
+        switch (incomeComparisonOperator.comparisonOperator) {
```
Comment (iamdiluxedbutcooler) [inline src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java:34]: If have time can consider making these greater/equal/less than behavior more type-safe (and more OOP) by implementing enums! Maybe something like this:

```
public enum IncomeComparisonOperator {
    EQUAL("="),
    GREATER_THAN(">"),
    LESS_THAN("<");
    // some assign symbol logic
}
```

File: src/test/java/seedu/address/model/util/IncomeComparisonOperatorTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class IncomeComparisonOperatorTest {
```
Comment (iamdiluxedbutcooler) [inline src/test/java/seedu/address/model/util/IncomeComparisonOperatorTest.java:12]: Very robust tests! You might want to add a test case for the `trim()` functionality in `isValidComparisonOperator ` method to double check too.

**Review by iamdiluxedbutcooler — APPROVED** [review]
LGTM! Very well-structured and comprehensive code pieces 👍🏻

File: src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java (hunk 2 above)
Comment (itsme-zeix) [reply src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java:25]: Great catch actually, I'll add that in.

Comment (itsme-zeix) [reply src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java:25]: Great idea, will implement it

**Pull request merged**